src/shogun/clustering/KMeans.cpp

```diff
@@ -167,6 +167,9 @@ void CKMeans::Lloyd_KMeans(SGMatrix<float64_t> centers, int32_t num_centers)
 				}
 			}
 		}
+
+		observe<SGMatrix<float64_t>>(iter, "mus");
```

Comment:
sorry, confused here. Why is this explicit observe call needed? Can't you just do "put". OR is this the same but more efficient?

Reply:
I used `observe()` here because it does not make sense to use `put()`. The method `Lloyd_KMeans` works directly on the `mus` variable, therefore I do not need to "put" it again (since it will be slower).

Reply:
for now, let's keep the `observe`. For the future: refactor this as `CIterativeMachine` which automatically shall emit model parameters